packages/react-router/modules/Redirect.js

```diff
@@ -49,6 +51,24 @@ class Redirect extends React.Component {
       this.perform()
   }
 
+  componentDidUpdate(prevProps) {
+    let prevTo = prevProps.to
+    let nextTo = this.props.to
+
+    prevTo = typeof prevTo === 'string' ? { pathname: prevTo } : prevTo
+    nextTo = typeof nextTo === 'string' ? { pathname: nextTo } : nextTo
+
+    if (locationsAreEqual(prevTo, nextTo))
+      return warning(
```

Comment:
The current code is essentially the same as `return console.error(...)`. While that _is_ legal, I think that it would be better to separate the call expression and return statement.
```js
if (...) {
  warning(...)
  return
}
```